proposals/esm-integration/README.md

```diff
@@ -88,57 +88,74 @@ Three things happen for each module during the Construction phase.
 
 ## Examples and explanation of import/export handling
 
 Some imports are challenging to handle in the current WebAssembly specification. In this section, we will discuss each kind of export and what the specification described above means for how it is handled.
 
 ### wasm imports <- JS exports
 
-| export type | value (not a WebAssembly.Global)* | global | memory | table | function |
-|-------------|------------------|--------|--------|-------|----------|
-|              | Error                         | Error  | Error  | Error | snapshot |
+When the Wasm module is evaluated, the exported value of the JS module is used in a way which is based on the import type declared in the WebAssembly module.
 
-\*While WebAssembly only has the concept of globals, JS could export either a regular JS value or a `WebAssembly.Global`.
+| import type | behavior |
+|-------------|----------|
+| global      | If the exported value is a WebAssembly.Global object, throw an exception if types mismatch, otherwise take that object as the import. Otherwise, if the imported type is `const`, cast the imported value to the appropriate numeric type, and create a new const global to hold the result. Otherwise, throw an exception. |
+| memory      | Check that the exported value is a WebAssembly.Memory object which meets the type imported; use it if so, otherwise, throw an exception |
+| table       | Check that the exported value is a WebAssembly.Table object which meets the type imported; use it if so, otherwise, throw an exception |
+| function    | If the exported value is a WebAssembly exported function based on the same types, make use of it. Otherwise, [create a host function](https://webassembly.github.io/spec/js-api/index.html#create-a-host-function) out of the JS function which includes the casts implied by its type. |
+
+While WebAssembly only has the concept of globals, JS could export either a regular JS value or a `WebAssembly.Global`.
 
 The sequence of operations for a wasm module which depends on a JS module is as follows:
 
 1. wasm module is parsed
 1. JS module is parsed
-1. JS module is instantiated. Functions are initialized. Non-function imports are set to undefined.
-1. wasm module is instantiated. A snapshot is taken of all imports. Functions are copied. All non-function imports are undefined and thus throw an error.
+1. JS module is instantiated. Function declaration exports are initialized. Other exports are set to undefined or are in TDZ.
+1. wasm module has an exports lexical environment created. Imports are bound to memory locations but values are not snapshotted yet.
+1. JS module is evaluated. All of its remaining exports (i.e. non-fuction values) are initialized. Any top-level statements are executed.
+1. wasm module is instantiated and its start function runs. All imports are snapshotted. All exports are initialized.
```

Comment:
All imports are snapshotted should be first, the start function could use them.

Reply:
Good point, will fix.